### PR TITLE
Migrate AvatarNetwork

### LINKED ISSUE #18884: Migrate components to TS: AvatarNetwork
### Description
Migrate `AvatarNetwork` component to TypeScript

### Technical Details
- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` follow [TS conventions](https://www.notion.so/Use-TypeScript-9ec7bd8fc8cf4d82b81cd8943c4707cd?pvs=4#7e6410b21100455a9a08442108f19d78)
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation 

### Acceptance Criteria
- Component has been converted to TypeScript

### PR DESCRIPTION
**Pull request by thebinij**
## Explanation

* Closes #18884 

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -11667,9 +11667,9 @@ __metadata:
...
 
 "caniuse-lite@npm:^1.0.30001109, caniuse-lite@npm:^1.0.30001400, caniuse-lite@npm:^1.0.30001426":
-  version: 1.0.30001431
-  resolution: "caniuse-lite@npm:1.0.30001431"
-  checksum: bc8ab55cd194e240152946b54bfaff7456180cc018674fc7ed134f4f502192405f6643f422feaa0a5e7cc02b5bac564cfac7771ac6d29f5d129482fcfe335ba1
+  version: 1.0.30001486
+  resolution: "caniuse-lite@npm:1.0.30001486"
+  checksum: 5e8c2ba2679e4ad17dea6d2761a6449b814441bfeac81af6cc9d58af187df6af4b79b27befcbfc4a557e720b21c0399a7d1911c8705922e38938dcc0f40b5d4b
   languageName: node
   linkType: hard
```
Comment (georgewrmarshall) [inline yarn.lock:11675]: There shouldn't be any updates to this file please revert

File: ui/components/ui/box/box.d.ts [hunk 2]
```diff
@@ -290,7 +290,7 @@ export interface BoxProps extends React.HTMLAttributes<HTMLElement> {
    * Use TEXT_ALIGN const from '../../../helpers/constants/design-system';
    * Accepts responsive props in the form of an array.
    */
-  textAlign?: BoxTextAlign | BoxTextAlignArray;
+  textAlign?: BoxTextAlign | BoxTextAlignArray | TextAlign;
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.d.ts:293]: I think I addressed this in a previous PR?

Comment (georgewrmarshall) [inline ui/components/ui/box/box.d.ts:293]: I think I addressed this in a previous PR? But we should update this file to use the new TextAlign enum and remove the TEXT_ALIGN usage and types

File: yarn.lock [hunk 3]
```diff
@@ -11667,9 +11667,9 @@ __metadata:
...
 
 "caniuse-lite@npm:^1.0.30001109, caniuse-lite@npm:^1.0.30001400, caniuse-lite@npm:^1.0.30001426":
-  version: 1.0.30001431
-  resolution: "caniuse-lite@npm:1.0.30001431"
-  checksum: bc8ab55cd194e240152946b54bfaff7456180cc018674fc7ed134f4f502192405f6643f422feaa0a5e7cc02b5bac564cfac7771ac6d29f5d129482fcfe335ba1
+  version: 1.0.30001486
+  resolution: "caniuse-lite@npm:1.0.30001486"
+  checksum: 5e8c2ba2679e4ad17dea6d2761a6449b814441bfeac81af6cc9d58af187df6af4b79b27befcbfc4a557e720b21c0399a7d1911c8705922e38938dcc0f40b5d4b
   languageName: node
```
Comment (georgewrmarshall) [inline yarn.lock:11673]: There shouldn't be any updates to this file please revert

File: ui/components/component-library/avatar-network/avatar-network.types.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  /**
+   * The showHalo accepts a boolean prop to render the image with halo effect
+   */
+  showHalo?: boolean;
+  /**
+   * The size of the AvatarNetwork
+   * Possible values could be AvatarBaseSize.Xs(16px), AvatarBaseSize.Sm(24px), AvatarBaseSize.Md(32px), AvatarBaseSize.Lg(40px), AvatarBaseSize.Xl(48px)
+   * Defaults to AvatarBaseSize.Md
+   */
+  size?: AvatarBaseSize;
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-network/avatar-network.types.ts:31]: Is there a way we can still create the `AvatarNetworkSize` type but allow AvatarBase to accept it?

File: shared/constants/network.ts [hunk 5]
```diff
@@ -3,28 +3,28 @@ import { capitalize, pick } from 'lodash';
...
 /**
  * Test networks have special symbols that combine the network name and 'ETH'
  * so that they are distinct from mainnet and other networks that use 'ETH'.
  */
 export type TestNetworkCurrencySymbol =
-  (typeof TEST_NETWORK_TICKER_MAP)[keyof typeof TEST_NETWORK_TICKER_MAP];
+  typeof TEST_NETWORK_TICKER_MAP[keyof typeof TEST_NETWORK_TICKER_MAP];
 
 /**
  * An object containing preferences for an RPC definition
```
Comment (georgewrmarshall) [inline shared/constants/network.ts:30]: I know this is just formatting but I think we don't want to make changes to this file. Could we revert them please?

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @thebinij, Thanks for your contribution. This is looking great. I have left some suggestions and a question.

File: ui/components/component-library/avatar-network/avatar-network.types.ts (hunk 4 above)
Comment (thebinij) [reply ui/components/component-library/avatar-network/avatar-network.types.ts:31]: One way is to allow a "size" property of AvatarBaseProps to accept both AvatarBaseSize | AvatarNetworkSize.

**Pull request closed without merging**